## Description: Reset default_adapter_id when there is no adapter.
## Origin/Author: Shih-Yuan Lee (FourDollars) <sylee@canonical.com>
## Bug: https://bugs.launchpad.net/ubuntu/+source/bluez/+bug/1303819
Index: trusty/src/manager.c
===================================================================
--- trusty.orig/src/manager.c	2014-04-08 10:26:15.040776000 +0800
+++ trusty/src/manager.c	2014-04-08 10:53:58.450958330 +0800
@@ -305,8 +305,10 @@
 	adapter_remove(adapter);
 	btd_adapter_unref(adapter);
 
-	if (adapters == NULL)
+	if (adapters == NULL) {
+		default_adapter_id = -1;
 		btd_start_exit_timer();
+	}
 }
 
 void manager_cleanup(DBusConnection *conn, const char *path)
